### ISSUE TITLE
AssertionError in FrameGrab.getFrame(File file,double sec) method

### ISSUE THREAD

**Issue by hxcode** [jcodec/jcodec#199]
`BufferedImage image = FrameGrab.getFrame(file, sec);`

the maven config :
```
  <dependency>
            <groupId>org.jcodec</groupId>
            <artifactId>jcodec-javase</artifactId>
            <version>0.1.9</version>
        </dependency>
```
stack trace : 
```
java.lang.AssertionError
	at org.jcodec.common.Assert.assertTrue(Assert.java:7)
	at org.jcodec.codecs.h264.decode.Intra8x8PredictionBuilder.predictWithMode(Intra8x8PredictionBuilder.java:27)
	at org.jcodec.codecs.h264.decode.SliceDecoder.decodeMBlockIntraNxN(SliceDecoder.java:869)
	at org.jcodec.codecs.h264.decode.SliceDecoder.decodeMBlockIInt(SliceDecoder.java:469)
	at org.jcodec.codecs.h264.decode.SliceDecoder.decodeMBlockI(SliceDecoder.java:461)
	at org.jcodec.codecs.h264.decode.SliceDecoder.decode(SliceDecoder.java:445)
	at org.jcodec.codecs.h264.decode.SliceDecoder.decode(SliceDecoder.java:266)
	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.decodeFrame(H264Decoder.java:87)
	at org.jcodec.codecs.h264.H264Decoder.decodeFrame(H264Decoder.java:61)
	at org.jcodec.api.specific.AVCMP4Adaptor.decodeFrame(AVCMP4Adaptor.java:71)
	at org.jcodec.api.FrameGrab.getNativeFrame(FrameGrab.java:259)
	at org.jcodec.api.awt.FrameGrab.getFrame(FrameGrab.java:83)
```
I have two mp4 video , I can get frame from one ,but another can't.
I use the tool ,MediaInfo, to analysis the details of mp4. Here is the successful-analysis mp4 info:
![20u axb9f 7sm m4baui k](https://user-images.githubusercontent.com/6957007/28010679-3e6262b2-6592-11e7-861a-ab328723c541.png)

below is the fail-analysis mp4  info:
![kd iu9m3r zuzsi9eaj8z_4](https://user-images.githubusercontent.com/6957007/28010508-c58ff084-6591-11e7-8843-261b5f3ca179.png)

**Comment by svitvitskiy** [comment]
Thank you for reporting. Could you upload a video that gives you that error?

**Comment by svitvitskiy** [comment]
I also noticed that you are using jcodec 0.1.9 which is a much older version. I would suggest that you try it with jcodec 0.2. If you find that the bug is still not fixed please upload the video and I'll try to look at it. Let me know.

**Comment by hxcode** [comment]
I have read the documentation from both the official website and README.md. They all suggested that I should use the version ***0.1.9***. In addition , I couldn't find the corresponding 0.2 version of the artifact on well-known maven repository website . But I also tried what you suggested me to do, here is the result : 

The  video ,[front_patch.zip](https://github.com/jcodec/jcodec/files/1137657/front_patch.zip),which is work with version 0.1.9 , but it doesn't work with version 0.2.0 now.
below is the version 0.2 stack trace : 
```
org.jcodec.api.UnsupportedFormatException: MPEG PS is temporarily unsupported.
	at org.jcodec.api.FrameGrab8Bit.createFrameGrab8Bit(FrameGrab8Bit.java:55)
	at org.jcodec.api.awt.AWTFrameGrab8Bit.createAWTFrameGrab8Bit(AWTFrameGrab8Bit.java:39)
	at org.jcodec.api.awt.AWTFrameGrab8Bit.getFrame(AWTFrameGrab8Bit.java:60)
```
I don't know why it tell me that exception message in version 0.2.0.

The  video ,[MUJI15S.zip](https://github.com/jcodec/jcodec/files/1137679/MUJI15S.zip), does not apply to version 0.1.9 and version 0.2.0 , below is the version 0.2.0 stack trace : 
```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.AssertionError

	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.waitForSure(H264Decoder.java:172)
	at org.jcodec.codecs.h264.H264Decoder$FrameDecoder.decodeFrame(H264Decoder.java:150)
	at org.jcodec.codecs.h264.H264Decoder.decodeFrame8BitFromNals(H264Decoder.java:100)
	at org.jcodec.codecs.h264.H264Decoder.decodeFrame8Bit(H264Decoder.java:96)
	at org.jcodec.api.specific.AVCMP4Adaptor.decodeFrame8Bit(AVCMP4Adaptor.java:83)
	at org.jcodec.api.FrameGrab8Bit.getNativeFrameWithMetadata(FrameGrab8Bit.java:224)
	at org.jcodec.api.awt.AWTFrameGrab8Bit.getFrameWithOrientation(AWTFrameGrab8Bit.java:91)
	at org.jcodec.api.awt.AWTFrameGrab8Bit.getFrame(AWTFrameGrab8Bit.java:60)
```

**Comment by svitvitskiy** [comment]
Ok, thanks, I'll have a look

**Comment by svitvitskiy** [comment]
A couple of things.

- The first video has been fixed and now it works in [JCodec 0.2.0](https://search.maven.org/#artifactdetails%7Corg.jcodec%7Cjcodec%7C0.2.0%7Cjar)
- The second video is interlaced, so it won't work, I have opened an issue #241 to work on the interlace support.
